app/src/main/java/org/glucosio/android/activity/AddGlucoseActivity.java

```diff
@@ -106,9 +110,9 @@ public void onNothingChosen(View labelledSpinner, AdapterView<?> adapterView) {
             }
         });
 
-        TextView unitM = (TextView) findViewById(R.id.glucose_add_unit_measurement);
+        TextView unitM = findViewById(R.id.glucose_add_unit_measurement);
 
-        if (presenter.getUnitMeasuerement().equals("mg/dL")) {
+        if (Constants.Units.MG_DL.equals(presenter.getUnitMeasurement())) {
```

Comment:
This entire logic could be moved and unit tested in isolation

Reply:
Yup, and again I'm not going to do anything with it